opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -2197,31 +2199,34 @@ namespace Opm {
         const int nupcol = this->schedule()[episodeIdx].nupcol();
         // don't switch group control when iterationIdx > nupcol
         // to avoid oscilations between group controls
+        // We also dont allow wells to change to group controls
+        // when iterationIdx > nupcol
```

Comment:
Should the opposite also apply, e.g., that wells under group control should not be allowed to change to individual in the subsequent `updateWellControl` (individual mode)?

Reply:
This change indeed reduce oscillations and improve convergence, but it has the potential to give very wrong results. I think we rather should remove the if statement completely i.e. always check the constrains. This will need some adjustment in the code to handle oscillatory controls for instance when a higher group has a constraint that is equal to the sum of the sub groups controls. I am testing out an option of detecting oscillatory controls like we do for the wells. This seems to work but need more testing. As a first move I plan to remove this part of this PR to be able to evaluate and test these things separately.

Reply:
OK, sounds good.